### ISSUE TITLE
Error Using Latest Dgraph 0.8 When Trying To Restart Dgraph After A Shutdown

### ISSUE THREAD

**Issue by willcj33** [dgraph-io/badger#125]
Below is the error. @manishrjain has more details as he walked me though it: 
```
willjaynes@dgraph-vm-0-8:/home/dgraph$ 2017/07/25 12:11:41 main.go:166: Loading configuration from file: config.yaml
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["max_memory_mb" = 13000]
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["trace" = 0.00]
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["p" = p]
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["w" = w]
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["debugmode" = false]
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["bindall" = true]
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["port" = 8080]
2017/07/25 12:11:41 init.go:75: Picked flag from config: ["grpc_port" = 9080]

Dgraph version   : v0.8.0
Commit SHA-1     : 49291a98
Commit timestamp : 2017-07-24 10:25:23 +1000
Branch           : HEAD

2017/07/25 12:11:41 Error while creating badger KV WAL store error: invalid argument
Unable to map file
github.com/dgraph-io/badger/y.Wrapf
	/home/travis/gopath/src/github.com/dgraph-io/badger/y/error.go:82
github.com/dgraph-io/badger/table.OpenTable
	/home/travis/gopath/src/github.com/dgraph-io/badger/table/table.go:148
github.com/dgraph-io/badger.newLevelsController
	/home/travis/gopath/src/github.com/dgraph-io/badger/levels.go:103
github.com/dgraph-io/badger.NewKV
	/home/travis/gopath/src/github.com/dgraph-io/badger/kv.go:199
github.com/dgraph-io/dgraph/dgraph.(*ServerState).initStorage
	/home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/server.go:69
github.com/dgraph-io/dgraph/dgraph.NewServerState
	/home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/server.go:54
main.main
	/home/travis/gopath/src/github.com/dgraph-io/dgraph/cmd/dgraph/main.go:688
runtime.main
	/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/proc.go:185
runtime.goexit
	/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/asm_amd64.s:2197
Opening table: "w/000003.sst"
github.com/dgraph-io/badger.newLevelsController
	/home/travis/gopath/src/github.com/dgraph-io/badger/levels.go:106
github.com/dgraph-io/badger.NewKV
	/home/travis/gopath/src/github.com/dgraph-io/badger/kv.go:199
github.com/dgraph-io/dgraph/dgraph.(*ServerState).initStorage
	/home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/server.go:69
github.com/dgraph-io/dgraph/dgraph.NewServerState
	/home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/server.go:54
main.main
	/home/travis/gopath/src/github.com/dgraph-io/dgraph/cmd/dgraph/main.go:688
runtime.main
	/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/proc.go:185
runtime.goexit
	/home/travis/.gimme/versions/go1.8.linux.amd64/src/runtime/asm_amd64.s:2197
```

**Comment by srh** [comment]
What sort of file system are you running Badger, or Dgraph, on? Are there any files of zero length in the directory?

**Comment by manishrjain** [comment]
Let's move to MANIFEST file -- get rid of compaction log, and updating an sst file's metadata in place. A more robust approach, also in use by RocksDB.

**Comment by bhuthesh** [comment]
@srh I also encountered the same error when running dgraph. What is the workaround or fix for it?

**Comment by manishrjain** [comment]
@bhuthesh : If the sst file is of length zero, then just delete it.

**Comment by manishrjain** [comment]
I have some cycles tomorrow, I'll pick this task up unless you already have a PR for this, @srh .

**Comment by manishrjain** [comment]
Have to work on something else. Thanks for taking it back over, @srh.

**Comment by srh** [comment]
Another bug in the compact log code is that we have the following:

```
	nextLevel.replaceTables(newTables) // will delete table files
	thisLevel.deleteTables(cd.top) // will delete table files

	// Write to compact log.
	c.done = 1
	s.clog.add(c)
```

What if we crash before writing the compact log?  We run into trouble, because we've deleted our old files and at startup we'll try to rollback the incomplete compaction by deleting the new ones.  We should have updated and synced the compact log *before* we deleted files.

**Comment by manishrjain** [comment]
I think we should interleave these. We tell the compact log that we have written the new ones, then delete the older ones (optionally writing to clog that we have deleted the older ones -- might not be required).

Steps:
- Write new tables
- Write to clog
- Delete old tables


That way, if we crash after:
- writing new tables. We'll delete them on restart.
- writing to clog. We'll use the new ones (and delete the old ones)
- deleting old tables. (everything's good).

**Comment by srh** [comment]
Was fixed with #140.
